### PR TITLE
[BC Breaking] Make sure the pipeline input shape is in WH order

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/yolox/yolox_s_8xb8-300e_coco.py [hunk 1]
```diff
@@ -83,7 +83,7 @@
     dict(
         type='RandomAffine',
         scaling_ratio_range=(0.1, 2),
-        border=(-img_scale[0] // 2, -img_scale[1] // 2)),
+        border=(-img_scale[1] // 2, -img_scale[0] // 2)),
```
Comment (ZwwWayne) [inline configs/yolox/yolox_s_8xb8-300e_coco.py:86]: should add comments for clarification.

File: configs/yolox/yolox_tiny_8xb8-300e_coco.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
     dict(
         type='RandomAffine',
         scaling_ratio_range=(0.5, 1.5),
-        border=(-img_scale[0] // 2, -img_scale[1] // 2)),
+        border=(-img_scale[1] // 2, -img_scale[0] // 2)),
```
Comment (ZwwWayne) [inline configs/yolox/yolox_tiny_8xb8-300e_coco.py:31]: should add comments for clarification.

File: mmdet/datasets/transforms/transforms.py [hunk 3]
```diff
@@ -630,7 +630,7 @@ class RandomCrop(BaseTransform):
 
     Args:
         crop_size (tuple): The relative ratio or absolute pixels of
-            height and width.
+            width and height.
```
Comment (ZwwWayne) [inline mmdet/datasets/transforms/transforms.py:633]: should reveal the order explicitly by indicating like `(width, height)`.

**Comment by ZwwWayne** [comment]
Please also update some doc in the repo.

File: configs/yolox/yolox_s_8xb8-300e_coco.py (hunk 1 above)
Comment (hhaAndroid) [reply configs/yolox/yolox_s_8xb8-300e_coco.py:86]: OK

**Comment by hhaAndroid** [comment]
> Please also update some doc in the repo to indicate our conventions.

I can't find any documentation for this information.

**Comment by BIGWangYuDong** [comment]
Should add some docs to tell users that the input setting is default as `W, H`, while the output should be `H, W`

File: docs/en/advanced_guides/conventions.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 Please check the following conventions if you would like to modify MMDetection as your own project.
 
+## Image pipeline input shape
+
+The image pipeline input shape is `(weight, height)`, and the shape of the pipeline output is `(height, weight)` in OpenMMLab 2.0.
```
Comment (ZwwWayne) [inline docs/en/advanced_guides/conventions.md:7]: Should be more specific and give several examples.
In model, the shape is also always (height, width)

File: docs/en/advanced_guides/conventions.md [hunk 5]
```diff
@@ -2,6 +2,38 @@
 
 Please check the following conventions if you would like to modify MMDetection as your own project.
 
+## Image pipeline input shape
+
+In OpenMMLab 2.0, the image pipeline input shape is `(width, height)`, the shape of the pipeline output is `(height, width)` and the shape is also always `(height, width)` in model. The common fields are as follows:
```
Comment (ZwwWayne) [inline docs/en/advanced_guides/conventions.md:7]: In OpenMMLab 2.0, to be consistent with the input argument of OpenCV, the argument about image shape in the data transformation pipeline is always in the `(width, height)` order. On the contrary, for computation convenience, the order of the field go through the data pipeline and model is `(height, width)`. The common fields are as below:

File: docs/en/advanced_guides/conventions.md [hunk 6]
```diff
@@ -2,6 +2,38 @@
...
+
+- img_shape: (height, width)
```
Comment (ZwwWayne) [inline docs/en/advanced_guides/conventions.md:9]: Specifically, in the results processed by each data transform pipeline, the fields and their value meaning is as below:

File: docs/en/advanced_guides/conventions.md [hunk 7]
```diff
@@ -2,6 +2,38 @@
...
+- ori_shape: (height, width)
+- pad_shape: (height, width)
+- batch_input_shape: (height, width)
+
+As an example, the initialization parameters of `Mosaic` are as follows:
```
Comment (ZwwWayne) [inline docs/en/advanced_guides/conventions.md:14]: On the contrary, the argument that build each pipeline is in the order of `(width, height)`, for example, the initialization arguments of `Mosaic` are as below:

File: docs/en/advanced_guides/conventions.md [hunk 8]
```diff
@@ -2,6 +2,38 @@
...
+       ...
+
+       # img_scale order should be (width, height)
+       self.img_scale = img_scale
+
+    def transform(self, results: dict) -> dict:
+        ...
+
+        results['img'] = mosaic_img
+        results['img_shape'] = mosaic_img.shape[:2]
```
Comment (ZwwWayne) [inline docs/en/advanced_guides/conventions.md:34]: is this (height, width) or (width, height)?

File: docs/en/advanced_guides/conventions.md [hunk 9]
```diff
@@ -2,6 +2,38 @@
 
 Please check the following conventions if you would like to modify MMDetection as your own project.
 
+## Image pipeline input shape
```
Comment (ZwwWayne) [inline docs/en/advanced_guides/conventions.md:5]: ## About the order of image shape

File: docs/en/advanced_guides/conventions.md (hunk 8 above)
Comment (hhaAndroid) [reply docs/en/advanced_guides/conventions.md:34]: (height, width)

**Pull request merged**